### DISCUSSION TITLE
Unexpected galois map index generated by GenLinearTransform(*)

### DISCUSSION THREAD

**Discussion by lilBuffaloEric** [tuneinsight/lattigo#334] [Q&A] [answered]
Hi contributors 

I found that in `ckks` GenLinearTransform(*), the index of `map[int]ringqp.Poly vec` will be set to `idx += slots ( if idx<0) or idx (if idex>=0)` . I guess this is doing `idx mod slots` . But I think this might potentially generate unexpected rotation index for me when I set logslots < logN -1, i.e. when number of slots required is exactly smaller than the maximum. For example, when `idx <0` , the galois element that GenLinearTransform generates is 5^(idx mod slots mod N/2) != 5^(idx mod N/2) , where 5^(idx mod N) is generated by  GenRotationKeysForRotations(*). Althought the two elements act in the same way, LinearTransformNew will panic when it can't find 5^(idx mod slots mod N) however 5^(idx mod N) is provided.

Not sure if my understanding is correct. But I did encountered this situation these days.

BR

**Comment by Pro7ech** [comment, marked as answer]
I have opened an issue.
